python/tank/bootstrap/cached_configuration.py

```diff
@@ -239,6 +241,10 @@ def update_configuration(self):
                     os.path.join(self._path.current_os, "install", "core")
                 )
                 log.debug("Previous core restore complete...")
+        else:
+            # remove backup folders now that the update has completed successfully
+            filesystem.delete_folder(os.path.dirname(config_backup_path))
```

Comment:
why are we deleting `os.path.dirname(config_backup_path)` and not just `config_backup_path`? Apologies if this wasn't clear in my previous feedback, but the point that i was trying to make was that i think we should only delete the stuff we create, e.g. `filesystem.delete_folder(config_backup_path)`.